recipes/boost/all/conanfile.py

```diff
@@ -1570,7 +1572,7 @@ def package_info(self):
 
             libformatdata = {}
             if not self.options.without_python:
-                pyversion = tools.Version(self._python_version)
+                pyversion = Version(self._python_version)
                 libformatdata["py_major"] = pyversion.major
                 libformatdata["py_minor"] = pyversion.minor
```

Comment:
This switches the code from using `conans.client.tools.version.Version` which wraps `semver` and has .major and .minor *properties* to `conans.model.version.Version` (which has .major and .minor *methods*).

So the later calls to .major and .minor aren't calling the method and we get:

```python
@(Pdb) Version(self._python_version).major
<bound method Version.major of '3.10'>
```
Should I revert the change or update these to

```python
pyversion.major(fill=False)
pyversion.minor(fill=False)
```



@uilianries wanted to check your intention before raising a bug.

Reply:
#12044

Reply:
@markferry The intention is only one: Move to Conan 2.0 https://docs.conan.io/en/latest/conan_v2.html Of course, it's a side-effect. Thanks for sharing, it deserves a new hotfix.